### PR TITLE
SSL_conf_cmd: add support for IgnoreUnexpectedEOF

### PR DESCRIPTION
**Pull request by sdaoden**
- [x] documentation is added or updated
(Trivial addition to existing array (not even compile tested, but should do))

### CODE REVIEW

**Comment by sdaoden** [comment]
P.S.: please do not bother cla stuff. Just take it -- i can post the patch on the ML instead!!  (sigh)

**Comment by tom-cosgrove-arm** [comment]
(They way copyright works they can't just "take it" if you post it on the mailing list - by far the easiest thing for all concerned would be `CLA: trivial`)

**Comment by sdaoden** [comment]
Thank you Tom!  I did how stated in the first paragraph..
Yes, well. Well.  All ISC, but only for the disclaimer that Public Domain does not have.
CLA: trivial is good.
Ciao.

**Comment by tom-cosgrove-arm** [comment]
Thanks! (It's easy to put a new work into the public domain, but a bit trickier with modifications, hence the CLA trivial thing - it just makes things more straightforward)

**Review by hlandau — APPROVED** [review]

**Comment by t8m** [comment]
This is borderline on CLA: trivial. @openssl/otc other members might have some opinion whether this is acceptable or not.

**Comment by t8m** [comment]
Also we should have a test for the feature, but I am setting `tests: deferred` as I assume Steffen would not be willing to write it (and sign regular CLA).

**Comment by sdaoden** [comment]
Hello again.

Thank you Tom!

Well t8m, greetings to Czechia.  I slowly get the impression novitschok or how
that weapon of mass-destruction is named that was used to poison people in GB
and that Nawalny was deliberately stolen from specifically Czechia, as only the
wonderful Miloš Zeman that unfortunately of all things now was scheduled to
leave office admitted.  1 of 500000000 or so Europeans to say the truth, oh
there are brave Czech people.  (Others unfortunately do not protect military
depots the way they should.)  But that is off-topic.  But so much propaganda you
know... and christianity has this "false testimony" commandment, but, well,
religion.  It is not the time.  Off-topic.

Well tests.  Right.  Oh, i do write a lot of tests!
I always had the impression that configuration files (oh! if they would all go
the route to allow a central /etc file to TLS configure all servers on a box!)
and SSL_conf_cmd are second-class citizens in OpenSSL, even though it is such
a fantastic and user friendly (if passed through to them) possibility!

I in fact "thought about" a script; ie, if new options are introduced one simply
fills in some thing with the constant, a conf_cmd name, documentation, and
a script which dispatches that to header, pod, and the file in this pull
request.  Like that it would all be bundled and step forward together.  And you
need to run perl to build OpenSSL anyhow.  But .. it surely will not happen.

Tests for that topic:

        #?0|kent:tls-openssl.git$ git grep SSL_OP_IGNORE_UNEXPECTED_EOF master -- test
        master:test/sslapitest.c: *         server does not set SSL_OP_IGNORE_UNEXPECTED_EOF,
        master:test/sslapitest.c: *         server sets SSL_OP_IGNORE_UNEXPECTED_EOF,
        master:test/sslapitest.c:        SSL_CTX_set_options(sctx, SSL_OP_IGNORE_UNEXPECTED_EOF);

        #?0|kent:tls-openssl.git$ git grep conf_cmd master -- test
        #?1|kent:tls-openssl.git$

Second class citizen.
I think it would only make sense -- from my point of view -- if sslapitest.c
would be rewritten to use all possibilities to set options, "somehow".
What you say?

Have a nice weekend everybody!

P.S.: oh please not!  Can a source change be any more trivial than this?

**Comment by sdaoden** [comment]
Ah, oh, wait:

#?0|kent:tls-openssl.git$ git grep CONF_cmd master -- test
master:test/README.ssltest.md:configurations. For details, see the manpages for `SSL_CONF_cmd`.
master:test/ssl_old_test.c:            rv = SSL_CONF_cmd_argv(c_cctx, &argc, &argv);
master:test/ssl_old_test.c:                rv = SSL_CONF_cmd_argv(s_cctx, &argc, &argv);
master:test/ssl_old_test.c:        rv = SSL_CONF_cmd(c_cctx, arg, argn);
master:test/ssl_old_test.c:            rv = SSL_CONF_cmd(s_cctx2, arg, argn);
master:test/ssl_old_test.c:                rv = SSL_CONF_cmd(s_cctx, arg, argn);
master:test/sslapitest.c:            if (!TEST_int_eq(SSL_CONF_cmd(confctx, "Options", "-AntiReplay"),
master:test/sslapitest.c:    if (!TEST_int_eq(SSL_CONF_cmd(cctx, "DHParameters", dhfile), 2))

But still..

**Comment by sdaoden** [comment]
I added a test; compile tested, but i cannot run it here due to
```
Can't locate Test/Builder/TodoDiag.pm in @INC (you may need to install the Test::Builder::TodoDiag module) (@INC contains: /x/src/tls-openssl.git/util/perl /home/steffen/src/symobj.git/lib /usr/lib/perl5/site_perl/5.36/linux-thread-multi /usr/lib/perl5/site_perl/5.36 /usr/lib/perl5/site_perl/5.36/linux-thread-multi /usr/lib/perl5/site_perl/5.36 /usr/lib/perl5/5.36/linux-thread-multi /usr/lib/perl5/5.36) at /usr/lib/perl5/5.36/Test/Builder.pm line 37.
```
And there is not easy package for CRUX Linux.  'Guess i could use cpan, but no cpan here yet, it installs too much (i do not need it!).  As this is super trivial i presume it works.  )If not i update the patchset..)

Sorry for the noise.

**Comment by sdaoden** [comment]
Oh heaven. Now it becomes work. Sorry for the noise, i'll be back when it works.

**Comment by sdaoden** [comment]
Ok (SSL_CONF_CTX_set_flags() i forgot).

So then on the day when Lenin's death is 99 years ago, he who finally failed last year to make friendship with the 30 percent Ukrainians who hated Russia for centuries, and who donated all that Russian land to the Ukrainians (less Crimea) for good brotherhood, even including all the black sea side that Russia now not even tries to get back, despite the cluster bombs and autonomous killing drones that were used.  But well.  This conflict was never about the people anyway.  And all of this is off-topic.

Thank you!  By the way, Congratulations!! for the QUIC series, it just came into the tree _that_ day!  I have not read it yet, but i have dim hopes that with OpenSSL one does not need any of those huge libraries that otherwise would be needed for QUIC!!  What an effort!
Congratulations again!

And a nice weekend i wish.

**Comment by tom-cosgrove-arm** [comment]
This is going to be tricky. It's great to have tests, but I suspect that the whole PR can no longer be classed as `CLA: trivial`. Since you've put this much effort into it, why not sign the CLA? It's honestly not that much work, compared to what you've done here!

**Comment by sdaoden** [comment]
That is saddening and mournful.

I cannot agree with the conclusion though since my only
real contribution is the word "IgnoreUnexpectedEOF",
and that, moreover, was not too far to seek.
Anything else is only copy+paste from

- Matt Caswell
  3bb5e5b09e32defefda2b61087c113203005ffa0
  "Add the ability to configure anti-replay via SSL_CONF"..
  For the SSL_CONF_CTX_* code in test/.

- Dmitry Belyavskiy
  09b90e0ed7915809fcd4ee1e250d881b77d06d45
  "Introducing option SSL_OP_IGNORE_UNEXPECTED_EOF"..
  For the manual.

- (again) Dmitry Belyavskiy
  e638112e15c63bfb4ab9bf5af66aa439e5983f23
  "Test for the SSL_OP_IGNORE_UNEXPECTED_EOF option"..
  His test function was flexible enough to support the
  simplemost possible addition that i have done.

- Maxim Mikityanskiy
  cd715b7e7fdd2aeb0fd80220d2df5187b291f87a
  "Add support for KTLS zerocopy sendfile on Linux"..
  For the clipboard snippet in ssl/ssl_conf.c that i used
  when adding the option.  (KTLSTxZerocopySendfile is one
  of the wildest camel etc case mixes i have ever seen.)

Now if that isn't CLA:trivial.
Hey you, from the company succeeding the one where Sophie Wilson
did, .. magic.  (Compare this!)
Better together!

P.S.:
Having read about CLA on Saturday, i think GnuPG allows signing
via a PGP-signed email to its development list.  That is so much
more friendly than what that PDF thing is about.
It is _such_ a pity that twenty years ago with the new passports
Germany did not also give out PGP and TLS certificates!  Is that
my identity or what???  Then again, even the Netherlands removed
their ROOT CA just a couple of weeks ago i saw it go from my
cacert.pem. .. Yes, 19th of December 2022.  Five days to Christmas.
Commercial mess!!

Ciao,

**Comment by tom-cosgrove-arm** [comment]
@mattcaswell @t8m Views on whether this is still `CLA: trivial`?

**Comment by mattcaswell** [comment]
Unfortunately I consider this to be beyond what we would accept under "CLA trivial". Please see here for details on our CLAs:

https://www.openssl.org/policies/cla.html

**Comment by sdaoden** [comment]
Someone with commit access might consider pulling this and committing it with --reset-author within the next days instead (this issue makes a long read already)?
This is the maximum trivialization i totally agree with!
Thank you.

**Comment by mattcaswell** [comment]
> Someone with commit access might consider pulling this and committing it with --reset-author within the next days instead

Unfortunately just changing the attribution for a commit doesn't solve the problem. All authors, regardless of whether they are listed as the author in the git commit, need to have a CLA on file.

**Comment by tom-cosgrove-arm** [comment]
If the first commit was (just about) acceptable with `CLA: trivial`, could just that commit be, with proper author attribution, be submitted on its own, leaving `tests: deferred`?

**Comment by mattcaswell** [comment]
> If the first commit was (just about) acceptable with CLA: trivial, could just that commit be, with proper author attribution, submitted on its own, leaving tests: deferred?

That first commit is probably borderline - in particular the documentation. Although I note that the text is largely a copy&paste of existing text from elsewhere in the man pages - so perhaps its ok. If it is agreed as trivial and submitted on its own, it could be accepted. Someone else could always write a test from scratch.

**Comment by t8m** [comment]
@sdaoden could you please drop the test commit from the PR if you aren't willing to submit the CLA? We can approve it then and somebody can rewrite the test from scratch later.

**Comment by sdaoden** [comment]
Hello.  Sorry for all the trouble.

Dear Matt Caswell.
No, it can all be replaced, committer _and_ author,
but for that you need to set the environment variables
as follows:

```
        GIT_COMMITTER_NAME='Matt Caswell' \
        GIT_COMMITTER_EMAIL=matt@openssl.org \
        GIT_AUTHOR_NAME=$GIT_COMMITTER_NAME \
        GIT_AUTHOR_EMAIL=$GIT_COMMITTER_EMAIL \
        git commit --amend --reset-author
```

For example -- and this is ONLY A TEMPORARY EXAMPLE!! -- i
did exactly this with the new version of this pull request
branch.  You can see it worked as expected!

Dear Tomáš Mráz.
I will tomorrow, but to show Matt Caswell that git(1) can,
in fact, replace both Author and Committer i today pushed
a merged commit with the above personalities.

I will (re-)push the mutilated test-less pull request with
my own identify and CLA:trivial again tomorrow, on your
request.

**Comment by mattcaswell** [comment]
> I will tomorrow, but to show Matt Caswell that git(1) can,
in fact, replace both Author and Committer i today pushed
a merged commit with the above personalities.

My point was not that it wasn't technically possible to do it (it is) - but that OpenSSL still requires a CLA from all authors no matter who git thinks the author is. Tricking the cla bot by listing someone else who does have a CLA but did not in actual fact write the commits doesn't make it an acceptable PR!

**Comment by t8m** [comment]
@sdaoden I realized you've dropped the commit but included the code in the first commit. That unfortunately won't work. Could you simply drop the code change in the test so it can be merged with CLA: trivial. We will reimplement a test independently.

**Comment by sdaoden** [comment]
Yes, here please.

**Review by t8m — APPROVED** [review]
OK with CLA: trivial

**Comment by t8m** [comment]
@mattcaswell @hlandau Please approve if you're OK with CLA: trivial.

**Review by mattcaswell — APPROVED** [review]

**Comment by mattcaswell** [comment]
On the basis that the text is largely a copy&paste from somewhere else, I agree that this is trivial.

**Comment by t8m** [comment]
Merged to master branch. Thank you for your contribution.

**Pull request closed without merging**
